### PR TITLE
add magnet control to triton driver

### PR DESCRIPTION
**Pull request by peendebak**
Control of the magnet is added.

@MerlinSmiles @jenshnielsen @WilliamHPNielsen The merge of the driver was not easy, and I cannot test all features here. Can you have a good look at the PR diff?

@gzheng @mnoordam

### CODE REVIEW

File: qcodes/instrument_drivers/oxford/triton.py [hunk 1]
```diff
@@ -141,11 +225,10 @@ def _get_control_channel(self):
                 self._control_channel = i
                 break
 
-        return self._control_channel
-
     def _set_control_channel(self, channel):
         self._control_channel = channel
-        self.write('SET:DEV:T{}:TEMP:LOOP:HTR:H1'.format(channel))
+        self.write('SET:DEV:T%s:TEMP:LOOP:HTR:H1' %
+                   self._get_control_channel())
```
[the remark this answers is not in the document]
Comment (peendebak) [reply qcodes/instrument_drivers/oxford/triton.py]: Probably someone not familar with `.format` style formatting

File: qcodes/instrument_drivers/oxford/triton.py [hunk 2]
```diff
@@ -141,11 +225,10 @@ def _get_control_channel(self):
                 self._control_channel = i
                 break
 
-        return self._control_channel
```
[the remark this answers is not in the document]
Comment (peendebak) [reply qcodes/instrument_drivers/oxford/triton.py]: Good catch

File: qcodes/instrument_drivers/oxford/triton.py [hunk 3]
```diff
@@ -199,16 +331,23 @@ def _get_temp_channel_names(self, file):
...
-        self.chan_temps = []
-        for i in range(1, 17):
-            chan = 'T%d' % i
-            self.chan_temps.append(chan)
-            self.add_parameter(name=chan,
-                               unit = 'K',
-                               get_cmd = 'READ:DEV:%s:TEMP:SIG:TEMP' % chan,
-                               get_parser = self._parse_temp)
-        self.chan_temps = set(self.chan_temps)
+    def _get_temp_channels(self, file):
```
[the remark this answers is not in the document]
Comment (peendebak) [reply qcodes/instrument_drivers/oxford/triton.py]: This PR is a merge for an old branch of qcodes. There indeed seem to be some changes not strictly related to the magnet

**Comment by peendebak** [comment]
@jenshnielsen I addressed some of your comments. I did not write the changes myself, so I cannot comment any further. 

@gzheng

**Comment by eendebakpt** [comment]
@jenshnielsen Should I do something with the codecov report?

**Comment by jenshnielsen** [comment]
@eendebakpt There are no tests for the driver hence the coverage stays the same as it is. No need to do anything

**Comment by peendebak** [comment]
@jenshnielsen What is the status of the PR?

**Comment by jenshnielsen** [comment]
It still have changes not related to the magnet that I don't  think belongs here

**Comment by peendebak** [comment]
@jenshnielsen I reverted the changes not related to the magnet. We will keep track of these changes locally.

**Comment by peendebak** [comment]
@jenshnielsen Is this one good to merge?

**Comment by jenshnielsen** [comment]
Yes it looks fine now

**Pull request merged**